tools/src/main/scala/scala/scalanative/build/NativeLib.scala

```diff
@@ -1,27 +1,95 @@
 package scala.scalanative
 package build
 
+import java.io.File
 import java.nio.file.Path
+import java.util.regex._
 
 /** Original jar or dir path and generated dir path for native code */
 private[scalanative] case class NativeLib(src: Path, dest: Path)
 
+/** Utilities for dealing with native library code */
 private[scalanative] object NativeLib {
+  private val fs           = File.separator
+  private val jarExtension = ".jar"
 
-  /** List of source patterns used */
-  val jarExtension  = ".jar"
-  val srcExtensions = Seq(".c", ".cpp", ".S", ".h", ".hpp")
-  val srcPatterns   = srcExtensions.mkString("glob:**{", ",", "}")
+  /** Object file extension: ".o" */
+  val oExt = ".o"
+
+  /** C++ file extension: ".cpp" */
+  val cppExt = ".cpp"
+
+  /** List of source patterns used: ".c, .cpp, .S" */
+  val srcExtensions = Seq(".c", cppExt, ".S")
+
+  /**
+   * Name of directory that contains native code: "scala-native"
+   */
+  val codeDir = "scala-native"
+
+  /** Used to find native source files in directories */
+  val srcPatterns = srcExtensions.mkString(
+    s"glob:**${fs}classes${fs}${codeDir}${fs}**{",
```

Comment:
The fact that `classes` is here is worrying. The tools shouldn't know about `classes`, which is sbt-specific. We should receive the classpath from the build tool, which would give us entries that end in `classes` if it's sbt, and look into those directories (for each one, in the subdirectory `scala-native`), but nowhere else.

Reply:
Right, I didn't like that either. Somehow I need to get the path to the `classes` dir no matter what it is called. Could we add that to `Config` so it is not essentially hard coded?

Reply:
You are correct, the classpath solves all the problems. Fixed.

Reply:
Just using the classpath worked. Thanks for the suggestion.